File: Frontend/src/app/service/backend/message.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { lastValueFrom } from 'rxjs';
import { environment } from '../../../environments/environments';
import { CreateMessage, Message, UUID } from '../../types';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  private baseUrl = `${environment.url}message`;

  constructor(private http: HttpClient) { }

  getMessageByRideId(id: UUID): Promise<Message[]> {
    return lastValueFrom(this.http.get<Message[]>(`${this.baseUrl}/${id}`)); 
  }

  sendMessage(command: CreateMessage) {
    lastValueFrom(this.http.post(`${this.baseUrl}`, command)); 
  }
}

File: Backend/src/main/java/com/example/r100/hop2school/service/MessageService.java
package com.example.r100.hop2school.service;



import com.example.r100.hop2school.dto.CreateMessage;
import com.example.r100.hop2school.dto.Message;

import java.util.List;
import java.util.UUID;

public interface MessageService {

    void sendMessage(CreateMessage command);

    List<Message> getMessagesForRide(UUID rideId);
}
